### ISSUE TITLE
Salary, salary_per_point, pay calculations

### ISSUE THREAD

**Issue by wingleungchan** [kevinhughes27/parity-server#137]
If I understand this correctly:
Pay = linear function of all the things you did during the game (goals, assists, ...)
Salary_per_point = {Pay for the game} / {Total points played in the game}

How is Salary calculated? Is it supposed to be pay normalized over 15 points? Salary_per_point * 15?

For example - Adam in Week 1:
      "d_points_against": 5, 
      "d_points_for": 6, 
      "o_points_against": 4, 
      "o_points_for": 3, 
      "pay": 100000, 
      "salary": 70643, 
      "salary_per_point": 5555.555555555556, 

salary_per_point = 100000 / (5 + 6 + 4 + 3) = 5555. this looks okay
but then 5555*15 = 83325 which does not match 70643

What am I missing?

**Comment by kevinhughes27** [comment]
The salary calculation always looks at all weeks because of this function https://github.com/kevinhughes27/parity-server/blob/master/server/models.py#L32. This is what I was trying to explain here https://github.com/kevinhughes27/parity-server/pull/136#discussion_r151163352. Things are a little tangled up and I am not sure what to do yet.

Also the salary is adjusted for the week here: https://github.com/kevinhughes27/parity-server/blob/master/server/app.py#L166

**Comment by kevinhughes27** [comment]
Something that might make sense but I didn't bother to do yet is to make the `average_salary_per_point` calculation https://github.com/kevinhughes27/parity-server/blob/master/server/models.py#L32 aware of what `week` it is dealing with. Then it would only load data from the past to calculate the average which would make more sense.

There is also too much logic in the `build_stats_response` function some of which we can address by coming at it from the other side and cleaning up how the UI displays data. Maybe we don't need an `All` tab and instead we can generate a separate response for summed statistics that doesn't need to be the same shape as the weekly data.

**Comment by kevinhughes27** [comment]
I did a thing which I think helps untangle this. In the PR where I split the models into their own files  (#160) I had to address a circular import dependency between stats and players. Stats used to have salary, salary_per_point and pay when only the last 2 where actually related to that stats object and salary was delegated to the player object.

With the change salary always comes from the player model and is merged into the stats response. Its not the cleanest solution yet but its moving in the right direction. 

I think my fix has closed the original confusion that lead to the issue so I am going to close.
